base/linalg/diagonal.jl

```diff
@@ -225,6 +225,13 @@ A_mul_B!(A::AbstractMatrix,B::Diagonal)  = scale!(A,B.diag)
 A_mul_Bt!(A::AbstractMatrix,B::Diagonal) = scale!(A,B.diag)
 A_mul_Bc!(A::AbstractMatrix,B::Diagonal) = scale!(A,conj(B.diag))
 
+Ac_mul_B(A::Diagonal, v::AbstractVector) = ctranspose.(A.diag) .* v
+At_mul_B(A::Diagonal, v::AbstractVector) = transpose.(A.diag) .* v
+
+A_mul_B!(vout::AbstractVector, A::Diagonal, vin::AbstractVector) = vout .= A * vin
```

Comment:
Couldn't this just be `vout .= A.diag .* vin` and thereby avoiding the allocation?